### PR TITLE
Resolved FastifyDeprecation warning and added typescript documentation.

### PR DESCRIPTION
**Pull request by vikasg603**
This pull request does
1) Resolve FastifyDeprecation warning, used https://github.com/SerayaEryn/fastify-session/pull/143
2) Added typescript documentation about the custom sessions.
3) Removed deprecated FastifyPlugin import and instead added FastifyPluginCallback.

### CODE REVIEW

File: lib/fastifySession.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ function session (fastify, options, next) {
   fastify.decorate('decryptSession', (sessionId, request, callback) => {
     decryptSession(sessionId, options, request, callback)
   })
-  fastify.decorateRequest('sessionStore', options.store)
-  fastify.decorateRequest('session', {})
+  fastify.decorateRequest('sessionStore', { getter: () => options.store })
+  fastify.decorateRequest('session', { getter: () => ({}) })
```
Comment (climba03003) [inline lib/fastifySession.js:22]:
```suggestion
  fastify.decorateRequest('session', { getter: () => ({}) })
```
You cannot change it with `getter` only. Please change it to `null` and add a empty object before or on `preValidation` hook.

Comment (vikasg603) [reply lib/fastifySession.js]: Easy fix.
My bad.

File: README.md [hunk 2]
```diff
@@ -119,6 +119,20 @@ fastify.decryptSession(sessionId, request, () => {
 })
 ```
 
+### Typescript support:
+This plugin support typescript, and you can extend fastify module to add your custom session type.
```
Comment (zekth) [inline README.md:123]:
```suggestion
This plugin supports typescript, and you can extend fastify module to add your custom session type.
```

**Review by zekth — CHANGES_REQUESTED** [review]
minor typo

**Review by vikasg603 — COMMENTED** [review]
Final review

File: lib/fastifySession.js [hunk 3]
```diff
@@ -18,8 +18,8 @@ function session (fastify, options, next) {
   fastify.decorate('decryptSession', (sessionId, request, callback) => {
     decryptSession(sessionId, options, request, callback)
   })
-  fastify.decorateRequest('sessionStore', options.store)
-  fastify.decorateRequest('session', {})
+  fastify.decorateRequest('sessionStore', { getter: () => options.store })
```
Comment (climba03003) [inline lib/fastifySession.js:21]: I am not sure about it. The `store` can be changed before but it cannot change now.

If it is `getter` only, the type should be changed to `readonly`.

Comment (vikasg603) [reply lib/fastifySession.js:21]: Hey @zekth, any comment on this ?

Comment (zekth) [reply lib/fastifySession.js:21]: Indeed this is a readonly value, please adjust the typings

Comment (vikasg603) [reply lib/fastifySession.js:21]: Hey, @zekth, I am not sure about what you are talking about?
Will you suggest to me the changes that need to do?

**Review by mcollina — APPROVED** [review]
lgtm, this is good

File: lib/fastifySession.js (hunk 3 above)
Comment (climba03003) [reply lib/fastifySession.js:21]: @vikasg603
Change it to readonly. https://github.com/fastify/session/blob/master/types/types.d.ts#L10
Add test-case inside https://github.com/fastify/session/blob/master/types/test/using-store.ts

It should be something like this, we expect it to be throwing error
```ts
// @ts-expect-error
request.sessionStore = null
```

**Review by climba03003 — APPROVED** [review]
LGTM.

**Review by zekth — APPROVED** [review]
lgtm! thanks

**Pull request merged**